src/ralph/configuration_management/views.py

```diff
@@ -0,0 +1,110 @@
+import logging
+
+from django.conf import settings
+from django.contrib.admin import FieldListFilter
+from django.core.exceptions import ObjectDoesNotExist
+from django.core.urlresolvers import NoReverseMatch
+from django.utils.safestring import mark_safe
+from django.utils.translation import ugettext_lazy as _
+
+from ralph.admin.helpers import get_admin_url
+from ralph.admin.views.extra import RalphDetailView
+from ralph.configuration_management.models import SCMScanStatus
+# NOTE(romcheg): These functions could be moved to a common place
+from ralph.security.views import _linkify, _url_name_for_change_view
+
+
+logger = logging.getLogger(__name__)
+
+
+class SCMScanInfo(RalphDetailView):
+
+    icon = 'cogs'
+    label = 'SCM info'
+    name = 'scm_info'
+    url_name = None
+    template_name = 'configuration_management/scminfo/scm_info.html'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        try:
+            scan = self.object.scmscan
+        except ObjectDoesNotExist:
+            scan = None
+        context['scm_scan'] = scan
+
+        hostname = getattr(self.object, 'hostname', None)
+
+        if hostname:
+            context['scm_tool_url'] = settings.SCM_TOOL_URL.format(
+                hostname=hostname
+            )
+
+        return context
+
+
+class SCMScanStatusInChangeListMixin(object):
+
+    icon_no_scan = '-'
+    icon_scan_ok = '<i class="fa fa-check-circle" aria-hidden="true"></i>'
+    icon_scan_error = ('<i class="fa fa-exclamation-triangle" '
+                       'aria-hidden="true"></i>')
+    icon_scan_fail = '<i class="fa fa-question-circle" aria-hidden="true"></i>'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.list_select_related += ['scmscan', ]
+        self.list_filter += [
+            (
+                'scmscan__scan_status',
+                self.custom_title_filter('SCM scan status')
+            ),
+            (
+                'scmscan__last_scan_date',
+                self.custom_title_filter('Last SCM scan date')
+            )
+        ]
+
+    # NOTE(romcheg): This could be probably moved to a common place.
+    def custom_title_filter(self, title):
+
+        class CustomTitledFilter(FieldListFilter):
+            def __new__(cls, *args, **kwargs):
+                filter_instance = FieldListFilter.create(*args, **kwargs)
+                filter_instance.title = title
```

Comment:
Because this filter must be an instance of FieldListFilter but in the upstream code they don't use __init__ to create objects. They use create() instead. I cannot return stuff from __init__ so I've used __new__.